lib/parse.js

```diff
@@ -2109,7 +2109,7 @@ function parse($TEXT, options) {
         return function_(AST_Accessor, is_generator, is_async);
     });
 
-    var object_or_object_destructuring_ = embed_tokens(function() {
+    var object_or_destructuring_ = embed_tokens(function object_or_destructuring_() {
```

Comment:
Ah - okay - if you need recursion it could be an issue. Feel free to drop the function expression name or rename it. I just needed any name to find the calling function.